File: practica2/practica2.java
package practica4;

import java.awt.*;
import javax.swing.*;

class Punto {
    private double x;
    private double y;

    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() { return x; }
    public double getY() { return y; }

    @Override
    public String toString() {
        return "Punto(x=" + x + ", y=" + y + ")";
    }
}

class Linea {
    private Punto p1;
    private Punto p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    @Override
    public String toString() {
        return "Linea(" + p1 + ", " + p2 + ")";
    }

    public void dibujaLinea(Graphics g) {
        g.drawLine((int) p1.getX() * 50, (int) p1.getY() * 50, (int) p2.getX() * 50, (int) p2.getY() * 50);
    }
}

class Circulo {
    private Punto centro;
    private double radio;

    public Circulo(Punto centro, double radio) {
        this.centro = centro;
        this.radio = radio;
    }

    @Override
    public String toString() {
        return "Circulo(Centro=" + centro + ", Radio=" + radio + ")";
    }

    public void dibujaCirculo(Graphics g) {
        int d = (int) (radio * 2 * 50);
        g.drawOval((int) (centro.getX() * 50 - d / 2), (int) (centro.getY() * 50 - d / 2), d, d);
    }
}

public class practica2 extends JPanel {
    private Linea linea;
    private Circulo circulo;

    public practica2(Linea linea, Circulo circulo) {
        this.linea = linea;
        this.circulo = circulo;
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        linea.dibujaLinea(g);
        circulo.dibujaCirculo(g);
    }

    public static void main(String[] args) {
        JFrame frame = new JFrame("Dibujar Línea y Círculo");
        Linea linea = new Linea(new Punto(1, 2), new Punto(4, 6));
        Circulo circulo = new Circulo(new Punto(3, 3), 2);
        
        System.out.println(linea);
        System.out.println(circulo);

        practica2 dibujo = new practica2(linea, circulo);
        frame.add(dibujo);
        frame.setSize(400, 400);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setVisible(true);
    }
}

File: practica2/practica2.py
import math
import matplotlib.pyplot as plt

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def __str__(self):
        return f"Punto(x={self.x}, y={self.y})"

class Linea:
    def __init__(self, p1: Punto, p2: Punto):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Linea({self.p1}, {self.p2})"

    def dibujaLinea(self):
        plt.plot([self.p1.x, self.p2.x], [self.p1.y, self.p2.y], 'bo-')
        plt.xlabel("X")
        plt.ylabel("Y")
        plt.title("Dibujo de Línea")
        plt.grid()
        plt.show()

class Circulo:
    def __init__(self, centro: Punto, radio: float):
        self.centro = centro
        self.radio = radio

    def __str__(self):
        return f"Circulo(Centro={self.centro}, Radio={self.radio})"

    def dibujaCirculo(self):
        fig, ax = plt.subplots()
        circulo = plt.Circle((self.centro.x, self.centro.y), self.radio, fill=False, edgecolor='r')
        ax.add_patch(circulo)
        ax.set_xlim(self.centro.x - self.radio - 1, self.centro.x + self.radio + 1)
        ax.set_ylim(self.centro.y - self.radio - 1, self.centro.y + self.radio + 1)
        ax.set_aspect('equal')
        plt.xlabel("X")
        plt.ylabel("Y")
        plt.title("Dibujo de Círculo")
        plt.grid()
        plt.show()


p1 = Punto(1, 2)
p2 = Punto(4, 6)
linea = Linea(p1, p2)
circulo = Circulo(Punto(3, 3), 2)

print(linea)  # Muestra la línea
linea.dibujaLinea()  # Dibuja la línea
print(circulo)  # Muestra el círculo
circulo.dibujaCirculo()  # Dibuja el círculo
